types/src/response.rs

```diff
@@ -88,40 +132,297 @@ pub struct SubscriptionPayloadError<'a, T> {
 	pub error: T,
 }
 
+/// Represent the result or error field of the JSON-RPC response object
+///
+/// It can be:
+///
+/// ```json
+/// "result":<value>
+/// "error":{"code":<code>,"message":<msg>,"data":<data>}
+/// ```
+#[derive(Debug, PartialEq)]
+pub enum PartialResponseSer<'a, T> {
```

Comment:
Nit: Would `ResponsePayloadSer` make more sense? (I guess I see the "result" or "error" as the payload of the response)